mongodbatlas/resource_mongodbatlas_search_index.go

```diff
@@ -0,0 +1,793 @@
+package mongodbatlas
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/go-test/deep"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/mwielbut/pointy"
+	"log"
+	"strings"
+
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+func resourceMongoDBAtlasSearchIndex() *schema.Resource {
+
+	return &schema.Resource{
+		Create: resourceMongoDBAtlasSearchIndexCreate,
+		Read:   resourceMongoDBAtlasSearchIndexRead,
+		Update: resourceMongoDBAtlasSearchIndexUpdate,
+		Delete: resourceMongoDBAtlasSearchIndexDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceMongoDBAtlasSearchIndexImportState,
+		},
+		Schema: returnSearchIndexSchema(),
+	}
+}
+
+func returnSearchIndexSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"project_id": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"cluster_name": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"index_id": {
+			Type:     schema.TypeString,
+			Computed: true,
+			Required: false,
+		},
+		"analyzer": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
```

Comment:
[q] why are we not providing a schema here?

Reply:
as the documentation here https://docs.atlas.mongodb.com/reference/api/fts-indexes-create-one/, the `analyzer` property is just a string.

the property `analyzers` (in plural) have an schema due is an array of objects